### ISSUE TITLE
Unable to do inference [deepmedic]

### ISSUE THREAD

**Issue by YilinLiu97** [NifTK/NiftyNet#80]
Hi, I trained a network using the deepmedic architecture, but could not use it for inference. 

I used this command: `python net_segment.py inference -c config/deepmedic_config.ini -a niftynet.application.segmentation_application.SegmentationApplication`

and got:
`INFO:niftynet: starting from iter -1
INFO:niftynet: set initial_iter to 139 based on checkpoints
INFO:niftynet: Accessing /home/yilinliu/NiftyNet-0.2.1/models/deepmedic/models/model.ckpt-139 ...
INFO:niftynet: Restoring parameters from /home/yilinliu/NiftyNet-0.2.1/models/deepmedic/models/model.ckpt-139
INFO:niftynet: grid sampling image sizes: {'image': (260, 215, 195, 1, 1)}
INFO:niftynet: grid sampling window sizes: {'image': (105, 105, 105, 1, 1)}
INFO:niftynet: yielding 1575756 locations from image, extended to 1575760 to be divisible by batch size 10
INFO:niftynet: Inference iter 0,  (12.719477s)
INFO:niftynet: Inference iter 1,  (3.273798s)
INFO:niftynet: Inference iter 2,  (3.281112s)
INFO:niftynet: Inference iter 3,  (3.277870s)
INFO:niftynet: Inference iter 4,  (3.284158s)
...`

The iteration seems to run forever and I haven't got the segmented output yet. 

Here is how I set up my config file:

`[TRAINING]
sample_per_volume = 50
rotation_angle = (-10.0, 10.0)
scaling_percentage = (-10.0, 10.0)
lr = 0.01
loss_type = CrossEntropy
starting_iter = 0
save_every_n = 3
max_iter = 138
max_checkpoints = 1000

[INFERENCE]
border = (100,100,100)
#inference_iter = 138
save_seg_dir = ~/NiftyNet/models/deepmedic/segmentation_output/
output_interp_order = 0
spatial_window_size = (105,105,105)

############################ custom configuration sections
[SEGMENTATION]
image = T1
label = parcellation
output_prob = False
num_classes = 11
label_normalisation = True`

Thanks!!

**Comment by mjorgecardoso** [comment]
You border size is way too long. The border variable, on the inference block, describes how much of the border should the network reject at each patch inference. At the moment you are rejecting too much and making the algorithm incredibly slow. Just set border to something much smaller, say (16,16,16).

**Comment by YilinLiu97** [comment]
@mjorgecardoso Thank you for the help! That problem was solved, but the output of the inference shows nothing.  Here is my setup:

`path_to_search = /home/yilinliu/ISMRM_Dataset/Training
filename_contains =
filename_not_contains =
spatial_window_size = (57,57,57)
pixdim = (1.0, 1.0, 1.0)
axcodes=(A, R, S)
interp_order = 3

[parcellation]
path_to_search = /home/yilinliu/ISMRM_Dataset/FullStructureLabels
filename_contains = _RIGHT_all_labels_8bit_path_RightLeftAmygdalaNOSUBFIELDS
filename_not_contains =
spatial_window_size = (9,9,9)
pixdim = (1.0, 1.0, 1.0)
axcodes=(A, R, S)
interp_order = 0

############################## system configuration sections
[SYSTEM]
cuda_devices = 0
num_threads = 2
num_gpus = 1
model_dir = ./models/deepmedic

[NETWORK]
name = deepmedic
activation_function = prelu
batch_size = 10
decay = 0
reg_type = L2

# volume level preprocessing
volume_padding_size = 12
# histogram normalisation
histogram_ref_file = ./example_volumes/monomodal_parcellation/standardisation_models.txt
norm_type = percentile
cutoff = (0.01, 0.99)
normalisation = False
whitening = False
normalise_foreground_only=True
foreground_type = otsu_plus
multimod_foreground_type = and

queue_length = 25
window_sampling = uniform

[TRAINING]
sample_per_volume = 1000
rotation_angle = (-10.0, 10.0)
scaling_percentage = (-10.0, 10.0)
lr = 0.01
loss_type = CrossEntropy
starting_iter = 0
save_every_n = 3
max_iter = 500
max_checkpoints = 1000

[INFERENCE]
border = (24,24,24)
#inference_iter = 186
save_seg_dir = ./segmentation_output/
output_interp_order = 0
spatial_window_size = (105,105,105)

############################ custom configuration sections
[SEGMENTATION]
image = T1
label = parcellation
output_prob = False
num_classes = 3
label_normalisation = True`
